### PR TITLE
Add support for undefining a domain

### PR DESCRIPTION
**Pull request by benlemasurier**

### CODE REVIEW

File: libvirt.go [hunk 1]
```diff
@@ -128,6 +128,20 @@ const (
 	MigrateFlagRDMAPinAll
 )
 
+// UndefineFlags specifies options available when undefining a domain.
+type UndefineFlags uint32
+
+const (
+	// UndefineFlagManagedSave removes all domain managed save data.
+	UndefineFlagManagedSave = 1 << iota
```
Comment (mdlayher) [inline libvirt.go]: I think you need to add the type explicitly for these to not just be of type `int`.

File: libvirt.go [hunk 2]
```diff
@@ -399,6 +413,42 @@ func (l *Libvirt) Run(dom string, cmd []byte) ([]byte, error) {
...
+// cleaning up snapshot metadata. For more information on available
+// flags, see UndefineFlag*.
+func (l *Libvirt) Undefine(dom string, flags UndefineFlags) error {
+	d, err := l.lookup(dom)
+	if err != nil {
+		return err
+	}
+
+	payload := struct {
+		Domain Domain
```
Comment (mdlayher) [inline libvirt.go]: No padding this time? :)

Comment (benlemasurier) [inline libvirt.go]: =p

File: libvirt.go (hunk 1 above)
Comment (benlemasurier) [inline libvirt.go]: oops, nice catch (it does work, but much better to be explicit)!

**Pull request merged**